### PR TITLE
Improve token transfer listing for tokens

### PR DESCRIPTION
**Pull request by csillag**
~**This depends on #743.**~

Improve pagination experience by proving rudimentary client-side pagination (limitation: looking only at the first 1000 records)

This impacts two pages:
- Tokens's token transfer listing (for example  `/mainnet/sapphire/token/0x998633BDF6eE32A9CcA6c9A247F428596e8e65d8` )
- Account's token transfer listing (for example `/mainnet/sapphire/address/0x998633BDF6eE32A9CcA6c9A247F428596e8e65d8/token-transfers#transfers`

~I think there is some problem with the pagination widget sometimes displaying the wrong number of total pages, but that is totally unrelated.~ **Update:** workaround added for anomalous behavior.

Technical note: great emphasis has been placed in developing this in separate commits so that it's easier to follow what is going on, so I suggest reviewing commit by commit.

### CODE REVIEW

File: src/app/pages/TokenDashboardPage/hook.ts [hunk 1]
```diff
@@ -41,10 +49,9 @@ export const useTokenTransfers = (scope: SearchScope, address: string) => {
     network,
     layer, // This is OK since consensus has been handled separately
     {
-      limit: NUMBER_OF_ITEMS_ON_SEPARATE_PAGE,
-      offset: offset,
+      ...pagination.paramsForQuery,
       rel: address,
-      // type: 'evm.log',
+      type: 'evm.log',
```
Comment (lukaw3d) [inline src/app/pages/TokenDashboardPage/hook.ts:54]: First commit should be a separate PR.
Also, can we combine useTokenTransfers and useAccountTokenTransfers now?

**Comment by csillag** [comment]
> First commit should be a separate PR.

OK, see #743

**Comment by csillag** [comment]
>  can we combine useTokenTransfers and useAccountTokenTransfers now?

Done.

File: src/app/components/Table/useClientSidePagination.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+const knownListKeys: string[] = ['total_count', 'is_total_count_clipped']
+
+function findListIn<T extends List, Item>(data: T): Item[] {
+  const candidates = Object.keys(data).filter(
+    key => !knownListKeys.includes(key) && Array.isArray(data[key as keyof T]),
+  ) as (keyof T)[]
+  switch (candidates.length) {
+    case 0:
+      throw new Error('Data not found!')
+    case 1:
```
Comment (lukaw3d) [inline src/app/components/Table/useClientSidePagination.ts:23]: With strict types this could be ensured at compile time, but not worth inventing it for this temporary workaround until backend adds filtering by signature

File: src/app/components/Table/useClientSidePagination.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    paramsForQuery,
+    getResults: (queryResult, key) => {
+      const data = queryResult
+        ? key
+          ? (queryResult[key] as Item[])
+          : findListIn<QueryResult, Item>(queryResult)
+        : undefined
+      const filteredData = !!data && !!filter ? data.filter(filter) : data
+
+      const wantedStart = (selectedClientPage - 1) * clientPageSize
```
Comment (lukaw3d) [inline src/app/components/Table/useClientSidePagination.ts:79]: I would use offset and limit for these calculations

**Review by lukaw3d — APPROVED** [review]
Looks easily reversible when we stop needing it

File: src/app/components/Table/useClientSidePagination.ts (hunk 3 above)
Comment (csillag) [reply src/app/components/Table/useClientSidePagination.ts]: Done.

**Comment by csillag** [comment]
> Looks easily reversible when we stop needing it

Yes, there is `useComprehensiveSearchParamsPagination()`, which implements the exact same interface as the client-side pagination engine, so it's a drop-in replacement.
